plotter/contour.go

```diff
@@ -0,0 +1,425 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:generate ./list
+
+package plotter
+
+import (
+	"math"
+	"sort"
+
+	"github.com/gonum/plot"
+	"github.com/gonum/plot/palette"
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// Contour implements the Plotter interface, drawing
+// a contour plot of the values in the GridXYZ field. The
+// order of rows is from high index to low index down.
+type Contour struct {
+	GridXYZ GridXYZ
+
+	// Levels describes the contour heights to plot.
+	Levels []float64
+
+	// LineStyle is the style of the contour lines.
+	draw.LineStyle
+
+	// Palette is the color palette used to render
+	// the heat map.
+	Palette palette.Palette
+
+	// Min and Max define the dynamic range of the
+	// heat map.
+	Min, Max float64
```

Comment:
Say you have the defaults, but you want to see which cotnoutrs are above some absolute threshold. You set the appropriate field or Min or Max and the (deleted but to be readded) fields Overflow and Underflow colours and they are drawn in the approriate colours.